docs/apispec-docs/src/main/scala/sttp/tapir/docs/apispec/schema/ToObjectSchema.scala

```diff
@@ -40,9 +41,14 @@ object ObjectSchema {
   }
 
   private def productSchemas[T](s: TSchema[T], st: TSchemaType.SProduct[T]): List[ObjectSchema] = {
-    (st.info -> s: ObjectSchema) +: fieldsSchema(st)
-      .flatMap(apply)
-      .toList
+    (st.info -> s: ObjectSchema) +: st.fields
+      .flatMap(a =>
+        a.schema match {
+          case s @ TSchema(_, _, _, _, _, _, _, _ @Validator.Enum(_, _, Some(info))) if referenceEnums(info) =>
```

Comment:
not sure what other validators we might have, but in theory we could have an `All` validator with two others, one of them being enum. So I think we'll have to extract the basic validators for the field, and then filter for only enum ones.

Reply:
Added by getting enum validators from all primitive ones. It's technically possible that there will be several enum validators for one field schema... so I look at the head of resulting collection, but maybe it should be validated or produce error?

Reply:
checking the first one sounds good :)